### ISSUE TITLE
isolateIds parameter does not work

### ISSUE THREAD

**Issue by gogolon** [dart-lang/tools#533]
Doc comment for the `collect` function states:
`If [isolateIds] is set, the coverage gathering will be restricted to only those VM isolates.`
This does not seem to be true. I'm trying to collect coverage from a single isolate, but the filter simply does not work and code executed on all isolates is included in the report.

**Reproduction**
1. Create a new dart project with `dart create isolate_coverage_sample`
2. Replace the code inside `bin/isolate_coverage_sample.dart` with the code below:
```dart
import 'dart:developer';
import 'dart:isolate';

import 'package:coverage/coverage.dart';
import 'package:isolate_coverage_sample/calculate.dart';
import 'package:isolate_coverage_sample/isolate_calculate.dart';

Future<void> main(List<String> arguments) async {
  calculate(); // This function is included by default in packages created using dart create

  final isolate = await Isolate.spawn(
    (_) => isolateCalculate(),
    null,
    paused: true,
  );
  final isolateId = Service.getIsolateId(isolate)!;
  final info = await Service.controlWebServer(enable: true);
  isolate.resume(isolate.pauseCapability!);

  final coverage = await collect(
    info.serverUri!,
    true,
    false,
    false,
    {'isolate_coverage_sample'},
    isolateIds: {isolateId},
  );

  print(coverage.entries);
}
```

3. Add a file `lib/isolate_calculate.dart` containing this code:
```dart
int isolateCalculate() {
  return 6 * 7;
}
```
4. While at the package's root, run:
`dart --pause-isolates-on-exit  bin/isolate_coverage_sample.dart`
5. Inspect the entries on the output. The file `package:isolate_coverage_sample/calculate.dart` is included in the coverage even though it was called on the main isolate.

**Comment by gogolon** [comment]
I thought that I have a use case, but maybe I was wrong. If coverage is collected after an isolate exits (but there are other isolates running), does the report contain coverage from the isolate that exited earlier? I see that `_getAllCoverage` iterates over all isolates and collects coverage for each of them. Therefore I assumed that if an isolate has already exited, its coverage will not be included in the report. I ran a quick test and the opposite seems to be true - the coverage from the exited isolate seems to be included. Is it related to isolate groups? Is it enough for just one isolate per group to be alive at the moment of collection in order to include coverage from all isolates that ever existed within the group? 

Due to my (false?) assumption, I was trying to pause the exiting isolates and collect coverage from them separately, using the `isolateIds` parameter.

**Comment by liamappelbe** [comment]
> Is it enough for just one isolate per group to be alive at the moment of collection in order to include coverage from all isolates that ever existed within the group?

Yeah, that makes sense to me. The coverage counters are shared between all isolates in the group, so even if one of the isolates has exited, the counters it was incrementing are still there.

**Comment by Victowolf** [comment]
@liamappelbe 
I have updated the documentation for the `collect` function to include details about isolate groups. The changes clarify how isolate groups are handled during coverage collection and provide better guidance for users.  

Let me know if any other changes have to me made.
